libmcount/trigger.c

```diff
@@ -0,0 +1,114 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <unistd.h>
+#include <string.h>
+#include <signal.h>
+#include <sys/time.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <inttypes.h>
+
+#include "libmcount/mcount.h"
+#include "utils/env-file.h"
+#include "utils/utils.h"
+
+
+#define ERROR_LIBRARY_NOT_EXIST \
+"%s does not exist libmcount shared object.\n"				\
+"\tmake sure that the file exists.\n\n"					\
+"\t[notification]\n"							\
+"\twhen you use dynamic tracing to the process, you must specify the\n"	\
+"\tabsolute path of 'libmcount-dynamic.so' not the relative path.\n\n" 	\
+"\tfor example:\n" 							\
+"\t[BADD]./uftrace dynamic -L. --pid=xxxx\n"				\
+"\t[GOOD]./uftrace dynaimc -L/home/m/uftrace  --pid=xxxx\n"
+
+#define ERROR_DATADIR_NOT_EXIST \
+"Cannot access to the %s for saving uftrace datas.\n"  			\
+"\tmake sure that the path exists.\n\n" 				\
+"\t[notification]\n"							\
+"\twhen you use dynamic tracing to the process, you must specify the\n"	\
+"\tabsolute path of data-dir.\n\n" 					\
+"\tfor example:\n" 							\
+"\t[BADD]./uftrace dynamic -L/home/m/uftrace --pid=xxxx\n" 		\
+"\t[GOOD]./uftrace dynaimc -L/home/m/uftrace"				\
+"--data=/home/m/uftrace/data --pid=xxxx\n"
+
+#define ERROR_ENVFILE_NOT_EXIST \
+"\tCannot access to the %s which contain environment variable.\n"
+
+
+__attribute__((weak)) void build_debug_domain(char *dbg_domain_str)
```

Comment:
I thought this `build_debug_domain()` can be duplicated with `build_debug_domain()` in libmcount/misc.c
So I thought we can include `internal.h` to this file, no ?

Reply:
@Taeung if it is possible, I would have done that. "internal.h" have many dependency with source file which located at libmcount. but libtrigger does not need other codes, need only 'build_debug_domain'. so, I have no choice. lately, I'll refactoring this code by decoupling 'build_debug_domain' from 'internal.h' if it need